emission/tests/modellingTests/TestGreedySimilarityBinning.py

```diff
@@ -24,31 +30,414 @@ def testBinning(self):
-        model_config = {
+        # pass in a test configuration to the binning algorithm.
+        #
+        # clustering_way : Part of the trip used for checking pairwise proximity.
+        #                  Can take one of the three values:
+        #                  
+        #                   1. 'origin' -> using origin of the trip to check if 2 points
+        #                                   lie within the mentioned similarity_threshold_meters
+        #                   2. 'destination' -> using destination of the trip to check if 2 points
+        #                                       lie within the mentioned similarity_threshold_meters
+        #                   3. 'origin-destination' -> both origin and destination of the trip to check 
+        #                                             if 2 points lie within the mentioned 
+        #                                              similarity_threshold_meters
+        
+        model1_config = {
             "metric": "od_similarity",
-            "similarity_threshold_meters": 500,  # meters,
+            "similarity_threshold_meters": 111,  # meters,
             "apply_cutoff": False,
+            "clustering_way": "origin",  
             "incremental_evaluation": False
         }
-        model = eamtg.GreedySimilarityBinning(model_config)
+        model1 = eamtg.GreedySimilarityBinning(model1_config)
+        model1.fit(trips)
+
+
+        model2_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters":111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "destination",
+            "incremental_evaluation": False
+        }
+        model2 = eamtg.GreedySimilarityBinning(model2_config)
+        model2.fit(trips)
+
+
+        model3_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "origin-destination",
+            "incremental_evaluation": False
+        }
+        model3 = eamtg.GreedySimilarityBinning(model3_config)
+        model3.fit(trips)
+
+        # Since neither the origin nor the destination of the points generated lie
+        # within the threshold, there should be no binning at all. All the bins should
+        # have size 1.
+
+        at_least_one_large_bin = all(map(lambda b: len(b['feature_rows']) == 1, model1.bins.values()))
+        self.assertTrue(at_least_one_large_bin, "no bin should have more than 1 features in it")
+
+        at_least_one_large_bin = all(map(lambda b: len(b['feature_rows']) == 1, model2.bins.values()))
+        self.assertTrue(at_least_one_large_bin, "no bin should have more than 1 features in it")
+
+        at_least_one_large_bin = all(map(lambda b: len(b['feature_rows']) == 1, model3.bins.values()))
+        self.assertTrue(at_least_one_large_bin, "no bin should have more than 1 features in it")
```

Comment:
nit: move the asserts just below the computation so that it is more clear what it is doing.

Reply:
will do.